homeassistant/components/tts/__init__.py

```diff
@@ -865,3 +870,60 @@ async def get(self, request: web.Request, filename: str) -> web.Response:
 def get_base_url(hass: HomeAssistant) -> str:
     """Get base URL."""
     return hass.data[BASE_URL_KEY] or get_url(hass)
+
+
+@websocket_api.websocket_command(
+    {
+        "type": "tts/engine/list",
+        vol.Optional("language"): str,
+    }
+)
+@callback
+def websocket_list_engines(
+    hass: HomeAssistant, connection: websocket_api.ActiveConnection, msg: dict
+) -> None:
+    """List text to speech engines and, optionally, if they support a given language."""
+    manager: SpeechManager = hass.data[DOMAIN]
+
+    language = msg.get("language")
+    providers = []
+    for engine_id, provider in manager.providers.items():
+        provider_info: dict[str, Any] = {"engine_id": engine_id}
+        if language:
+            provider_info["language_supported"] = bool(
+                language_util.matches(language, provider.supported_languages)  # type: ignore[arg-type]
```

Comment:
The type ignore can be removed if #91347 is merged

Reply:
This has been merged. Rebase possible.